src/include/miopen/mha/invoke_params.hpp

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/invoke_params.hpp>
+#include <miopen/tensor.hpp>
+#include "mha.hpp"
+
+namespace miopen {
+namespace mha {
+
+struct InvokeParams : public miopen::InvokeParams
```

Comment:
minor suggestion: rename to `MhaInvokeParams` to be inline with other derived classes of `InvokeParams`.

Reply:
Well, I 'm looking at other primitives....
softmax - just "InvokeParams" class under softmax namespace.
pooling  - "FwdInvokeParams" class under pooling namespace
reduce - "InvokeParams" class under reduce namespace,
etc

Reply:
I guess then we're not consistent in our repo. We have `WrwInvokeParams` and `DataInvokeParams` for convolution and then fusion stuff is named similarly I believe.

Reply:
I am of the opinion that we should name the classes explicitly instead of relying on what namespace a certain `InvokeParams` is in and sometimes it may also lead to collisions that are not detected. 
CC: @atamazov , @DrizztDoUrden

Reply:
@amberhassaan 
What kind of non-detected collisions do you mean?
If we do not overuse `using namespace` - namespace is a good solution.